parsing/intermediate/dialects/dialect1/parserImpls.ftl

```diff
@@ -963,11 +964,166 @@ SqlCreate SqlCreateTable() :
             }
         }
     ]
+    [
```

Comment:
Would this be able to parse a statement that has a "partition by" in between two indices? It looks like it's currently only parsed after the indices are parsed but the docs show in can occur in between or before indices as well.

Reply:
Yeah you are right. I was looking through the documentation trying to find your comment about putting it somewhere in the SqlCreateTableIndex(), but can't find that reference and got convinced that Partition is not part of the indices. I just double checked the documentation, and will make the correction soon.

Reply:
Yeah I'm not sure if partition by is considered an index, but it might still be able to be used as a SqlCreateTableIndex since they have similar parameters like a SqlNodeList? Or it could be parsed in an OR statement with the SqlCreateTableIndex() calls above if it doesn't fit with SqlCreateTableIndex

Reply:
I think it would be good to have partition parse in an OR statement with the SqlCreateTableIndex() calls. I've added the parsing logic and some test in https://github.com/googleinterns/calcite/pull/219/commits/fd9845d21c594de4850f21d984dad9c190cc19d2. Will update the design doc to reflect the changes.